hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/common/model/HoodieSparkRecord.java

```diff
@@ -440,7 +443,9 @@ private static void validateRow(InternalRow data, StructType schema) {
     //       corresponding schema has to be provided as well so that it could be properly
     //       serialized (in case it would need to be)
     boolean isValid = data == null || data instanceof UnsafeRow
-        || schema != null && (data instanceof HoodieInternalRow || SparkAdapterSupport$.MODULE$.sparkAdapter().isColumnarBatchRow(data));
+        || schema != null && (data instanceof HoodieInternalRow
+        || data instanceof GenericInternalRow
```

Comment:
The original indentation is more clear.

Reply:
The line is too long.  I've revised the indentation to make it more clear.